tests/integration/Migration/Action/GenerateCest.php

```diff
@@ -216,4 +216,68 @@ public function getQuoteWrappedColumns(IntegrationTester $I): void
         $I->assertSame("'column1'", $class->getQuoteWrappedColumns()[0]);
         $I->assertSame("'column2'", $class->getQuoteWrappedColumns()[1]);
     }
+
+    /**
+     * @param IntegrationTester $I
+     */
+    public function throwUnknownColumnTypeException(IntegrationTester $I): void
+    {
+        $I->wantToTest('Migration\Action\Generate - getColumns() throw UnknownColumnTypeException');
+
+        $I->expectThrowable(UnknownColumnTypeException::class, function () {
+            $columns = [
+                new Column('unknown', [
+                    'type' => 9000,
+                    'size' => 10,
+                    'notNull' => true,
+                ]),
+            ];
+
+            $data = [];
+            $class = new Generate('mysql', $columns);
+            foreach ($class->getColumns() as $column) {
+                // Wait error
+                $data[] = $column;
+            }
+        });
+    }
+
+    /**
+     * @param IntegrationTester $I
+     * @throws UnknownColumnTypeException
+     */
+    public function columnHasDefault(IntegrationTester $I): void
+    {
+        $I->wantToTest('Migration\Action\Generate - getColumns() has default');
+
+        $expected = "'default' => \"0\"";
+        $columnsWithDefault = [
+            new Column('column_default', [
+                'type' => Column::TYPE_INTEGER,
+                'size' => 10,
+                'notNull' => true,
+                'default' => 0,
+            ]),
+        ];
+
+        $columnsWithDefaultAndAI = [
+            new Column('column_ai', [
+                'type' => Column::TYPE_INTEGER,
+                'size' => 10,
+                'notNull' => true,
+                'autoIncrement' => true,
+                'first' => true,
+                'primary' => true,
+                'default' => 0,
+            ]),
+        ];
+
+        $class1 = new Generate('mysql', $columnsWithDefault);
```

Comment:
Same things

Reply:
Here it is not DB test, but class(es) logic.